test/e2e/network/networking_perf.go

```diff
@@ -18,145 +18,275 @@ package network
-		nodes, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
-		framework.ExpectNoError(err)
-		totalPods := len(nodes.Items)
-		// for a single service, we expect to divide bandwidth between the network.  Very crude estimate.
-		expectedBandwidth := int(float64(maxBandwidthBits) / float64(totalPods))
-		appName := "iperf-e2e"
-		_, err = e2eservice.CreateServiceForSimpleAppWithPods(
-			f.ClientSet,
-			8001,
-			8002,
-			f.Namespace.Name,
-			appName,
-			func(n v1.Node) v1.PodSpec {
-				return v1.PodSpec{
-					Containers: []v1.Container{{
-						Name:    "iperf-server",
-						Image:   imageutils.GetE2EImage(imageutils.Agnhost),
-						Command: []string{"/bin/sh"},
-						Args: []string{
-							"-c",
-							"/usr/local/bin/iperf " + familyStr + "-s -p 8001 ",
+	if isIPV6 {
+		args = append(args, "-V")
+	}
+	deploymentSpec := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "iperf2-server-deployment",
+			Labels: labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					TerminationGracePeriodSeconds: &one,
+					Containers: []v1.Container{
+						{
+							Name:    "iperf2-server",
+							Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+							Command: []string{"iperf"},
+							Args:    args,
+							Ports: []v1.ContainerPort{
+								{
+									ContainerPort: iperf2Port,
+									Protocol:      v1.ProtocolTCP,
+								},
+							},
 						},
-						Ports: []v1.ContainerPort{{ContainerPort: 8001}},
-					}},
-					NodeName:      n.Name,
-					RestartPolicy: v1.RestartPolicyOnFailure,
-				}
+					},
+				},
 			},
-			// this will be used to generate the -service name which all iperf clients point at.
-			numServer, // Generally should be 1 server unless we do affinity or use a version of iperf that supports LB
-			true,      // Make sure we wait, otherwise all the clients will die and need to restart.
-		)
+		},
+	}
 
-		if err != nil {
-			framework.Failf("Fatal error waiting for iperf server endpoint : %v", err)
-		}
+	deployment, err := client.AppsV1().Deployments(namespace).Create(context.TODO(), deploymentSpec, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("deployment %q Create API error: %v", deploymentSpec.Name, err)
+	}
+	framework.Logf("Waiting deployment %q to complete", deploymentSpec.Name)
+	err = e2edeployment.WaitForDeploymentComplete(client, deployment)
+	if err != nil {
+		return nil, fmt.Errorf("deployment %q failed to complete: %v", deploymentSpec.Name, err)
+	}
+	return deployment, nil
+}
+
+func iperf2ServerService(client clientset.Interface, namespace string, labelKey string, labelValue string) (*v1.Service, error) {
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: labelValue},
```

Comment:
this is misleading, it took me a while to realise how iperf clients connects to the service, and is because the service name has the labelValue as name, hence, labelValue has a DNS record pointing to the service IP.
Also, this is no consistent with the other objects created (deployments, pods, containers, ... ) where you are hardcoding the names.

https://github.com/kubernetes/kubernetes/pull/94015/files#diff-c70b0afa39c9d7108ad9cd29eee4a49894807e5044eb82b348919109b6ae8b6eR261

Reply:
How would you like this to be organized?  I'm fine with anything 👍

Reply:
maybe we can declare the `Names` as constants?

Reply:
👍 yep sounds good !!

Reply:
sorry, it seems I didn't explain myself, I didn't mean to replace label key and values, just the `Name` field.
So, in this case, the `Service` `Name` is the same that the `labelValue`, that is the part that is misleading.
I was suggesting to use a new const `iperfServiceName` or something like that and use it to configure the `Service` and to use it as the `iperf` client destination https://github.com/kubernetes/kubernetes/pull/94015/files#diff-c70b0afa39c9d7108ad9cd29eee4a49894807e5044eb82b348919109b6ae8b6eR261

Reply:
ahh 👍 got it, yeah that makes sense!  🦖 🦖